### ISSUE TITLE
Custom webpack config doesnt allow importing static assets from css

### ISSUE THREAD

**Issue by sami616** [vercel/next.js#33781]
### Run `next info` (available from version 12.0.8 and up)

 Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 21.1.0: Wed Oct 13 17:33:23 PDT 2021; root:xnu-8019.41.5~1/RELEASE_X86_64
    Binaries:
      Node: 14.18.1
      npm: 8.1.3
      Yarn: 1.22.5
      pnpm: 6.23.2
    Relevant packages:
      next: 12.0.9
      react: 18.0.0-rc.0-next-fe905f152-20220107
      react-dom: 18.0.0-rc.0-next-fe905f152-20220107

### What version of Next.js are you using?

12.0.9

### What version of Node.js are you using?

14.18.1

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

next export

### Describe the Bug

My understanding is that nextjs doesnt allow importing of static assets from global css. The docs say to put assets into a public dir and reference from there.

Because of reasons, the above solution isnt an option for me and i need to be able to reference imported static assets relatively from my css files.

My attempt at configuring webpack to allow asset rebasing results in a the assets paths in my css being prepended with `webpack:///mini-css-extract-plugin/`

Something like the below is output:

`url(webpack:///mini-css-extract-plugin/5fSGMdx82yH3OK2WXjGYzS/_next/static/media/better-times-for-bp.84838048.woff2)`

The custom webpack config in my `next.js.config` looks like this:

```
const config = {
  reactStrictMode: true,
  experimental: { externalDir: true, topLevelAwait: true },
  webpack(config, options) {
    config.module.rules.push({
      test: /\.(png|jpg|gif|svg|eot|ttf|woff|woff2)$/,
      use: [
        {
          loader: 'url-loader',
          options: {
            limit: 100000
          }
        }
      ]
    })
    return config
  }
}
```

### Expected Behavior

Assets are copied to the build directory and correctly referenced from css files.

### To Reproduce

1: Create a css file that references an asset in your project using a relative path: ie `background: ur('../../my-bg.jpg')`
2: Import that css in your _app.js
3: Run next build
4: Run next export

**Comment by sami616** [comment]
Related discussion: https://github.com/vercel/next.js/discussions/33094

**Comment by mikaelbr** [comment]
Had this issue in production for some time without noticing (we're using a design system as a separate npm module containing font files, and as I've got the fonts installed locally I haven't noticed the issue until recently).

```css
/* app.css */

@import '~@some-package/typography/index.css';
```

```css
/* @some-package/typography/index.css */

@font-face {
  font-family: 'Graphik Web';
  src: url('./fonts/graphik/Graphik-Regular-Web.woff2') format('woff2'),
    url('./fonts/graphik/Graphik-Regular-Web.woff') format('woff');
  font-weight: 400;
  font-style: normal;
  font-stretch: normal;
  font-display: swap;
}
```

Where the font is distributed using the design system. This way we keep multiple sites consistent without having to manually copy or provide any copy-files configuration and additional build steps.

This is something that worked before, and I'd say at least at v10. There are no specific webpack loaders for woff files. Only plugin in use is `next-images`. No overrides as the original issue here, just plain old Next.js config in regards to assets.

Next version: `next@12.1.0` tested with canary without any difference.

Happy to provide a repro if necessary and help with debugging/fixing if needed.

**Comment by sami616** [comment]
@mikaelbr yes I have just realised it's only an issue when trying to import the font from a separate local package. We're using a monorep with our local font inside a 'shared' package, so almost the exact same issue you're facing

**Comment by mikaelbr** [comment]
Haven't confirmed yet. Seems not unreasonable that there's a bug introduced by https://github.com/vercel/next.js/pull/30617 or https://github.com/vercel/next.js/pull/30743. Requires compiling/building vendors and replacing to verify. I'll try to look more into it tomorrow and see if I can provide a fix.

**Comment by sami616** [comment]
I did a bit more digging today, 

My css file is made up of two things:

1. `@import` for a google font
2. `@font-face` for a local font

When i remove the `@import`, the font-face starts working. 

Running `12.0.9`

Very odd 🤔

**Comment by mikaelbr** [comment]
@sami616 Maybe change the issue title to be more descriptive? Haven't found a clear solution yet, but to me it looks like after 12.0.3 no relative assets are supported from CSS files through file-resolver/asset generator. Even without custom config.

**Comment by mikaelbr** [comment]
Still not working in `next@12.1.1-canary.5`. I have a feeling it's an issue with `mini-css-extract-plugin`. But due to the vendor bundling and ncc not producing the same output/module export as existing I'm having a hard time checking if the latest `mini-css-extract-plugin` fixes the issue.

**Comment by sokra** [comment]
> I have a feeling it's an issue with `mini-css-extract-plugin`.

Same.

It seem to be caused by the import to the external css `@import url("//hello.myfonts.net/count/3a95ac");`. That seem to break the rest of the file...

As workaround you can put the external `@import` into a separate file and `@import` that.

cc @alexander-akait

**Comment by mikaelbr** [comment]
You're right. Looks like there's an error parsing/using schemaless URIs that breaks the entire file or something. Verified that moving it to a separate file works. Thanks.

**Comment by keithtxw** [comment]
Hi guys I am having the same scenario as @mikaelbr, in which my fonts are in our design system, consume as a node module. But my error is as such:
```
Access to font at 'webpack:///mini-css-extract-plugin/<app_name>/_next/static/media/Lato-Regular.eced59bb.ttf' from origin 'http://localhost:3000' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, chrome-untrusted, https.
```

Are you guys encountering something similar as well?

**Comment by JanKaifer** [comment]
I understood that this is not next specific issue and there was no activity for a while.
